components/Search.js

```diff
@@ -24,23 +24,14 @@ class Search extends Component {
 
   state = {
     format: 'list',
-    value: '',
-    hasContent: false,
   };
 
   toggleClearIcon = evt => {
-    const hasContent = evt.target.value.length > 0;
-    this.setState({
-      hasContent,
-      value: evt.target.value,
-    });
+    this.input.value = evt.target.value;
```

Comment:
Do you think we could also add a `this.props.onInput()` call here? That way you can trigger more stuff on input instead of just clearing the icon (implementing a search on carbondesignsystem.com right now, and found that I needed this).